From linux-usb-owner@vger.kernel.org Mon Aug 11 09:28:31 2008
From: Robert Jarzmik <robert.jarzmik@free.fr>
Date: Mon, 11 Aug 2008 18:28:13 +0200
Subject: USB: Fix pxa27x_udc usb speed handling.
To: linux-usb@vger.kernel.org
Cc: david-b@pacbell.net, Robert Jarzmik <robert.jarzmik@free.fr>
Message-ID: <1218472093-18732-1-git-send-email-robert.jarzmik@free.fr>


The new composite framework revealed a weakness in the
pxa27x_udc driver gadget register function. Instead of
checking if speed asked for was USB_LOW_SPEED upon
usb_gadget_register() to deny service, it checked only
for USB_FULL_SPEED, thus denying service to usb high
speed capable gadgets (like g_ether).

Signed-off-by: Robert Jarzmik <robert.jarzmik@free.fr>
Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/pxa27x_udc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/gadget/pxa27x_udc.c
+++ b/drivers/usb/gadget/pxa27x_udc.c
@@ -1622,7 +1622,7 @@ int usb_gadget_register_driver(struct us
 	struct pxa_udc *udc = the_controller;
 	int retval;
 
-	if (!driver || driver->speed != USB_SPEED_FULL || !driver->bind
+	if (!driver || driver->speed < USB_SPEED_FULL || !driver->bind
 			|| !driver->disconnect || !driver->setup)
 		return -EINVAL;
 	if (!udc)
